src/chainer.test.js

```diff
@@ -2,0 +1,29 @@
+"use strict";
+
+describe(`chainer`, () => {
+  const { chainer } = require("./chainer");
+
+  it(`should correctly chain and execute functions with arguments`, () => {
+    const f1 = jest.fn((x) => x * 2);
+    const f2 = jest.fn((x) => x + 2);
+    const f3 = jest.fn((x) => Math.pow(x, 2));
+
+    chainer(f1, f2, f3);
+
+    expect(f1(2)).toEqual(4);
+    expect(f2(4)).toEqual(6);
+    expect(f3(1)).toEqual(1);
```

Comment:
These expectations are testing the individual functions directly rather than the result of the chained function. You should instead test the result of the chained function, e.g., `expect(chainedFunction(2)).toEqual(expectedResult)`.